saleor/plugins/email_common.py

```diff
@@ -0,0 +1,386 @@
+                )
+            }
+        )
+
+    try:
+        validate_email_config(config)
+    except Exception as e:
+        logger.warning("Unable to connect to email backend.", exc_info=e)
+        error_msg = (
+            "Unable to connect to email backend. Make sure that you provided "
+            "correct values."
+        )
+        raise ValidationError(
+            {
+                c: ValidationError(
+                    error_msg, code=PluginErrorCode.PLUGIN_MISCONFIGURED.value
+                )
+                for c in configuration.keys()
+            }
+        )
+
+
+def validate_format_of_provided_templates(
+    plugin_configuration: "PluginConfiguration", template_fields: List[str]
+):
+    """Make sure that the templates provided by the user have the correct structure."""
+    configuration = plugin_configuration.configuration
+    configuration = {item["name"]: item["value"] for item in configuration}
+
+    if not plugin_configuration.active:
+        return
+    compiler = pybars.Compiler()
+    errors = {}
+    for field in template_fields:
+        template_str = configuration.get(field)
+        if not template_str or template_str == DEFAULT_EMAIL_VALUE:
+            continue
+        try:
+            compiler.compile(template_str)
+        except pybars.PybarsError:
+            errors[field] = ValidationError(
+                "The provided template has an inccorect structure.",
+                code=PluginErrorCode.INVALID.value,
+            )
+    if errors:
+        raise ValidationError(errors)
+
+
+def get_email_template(
+    plugin_identifier: str, template_field_name: str, default: str
+) -> str:
+    """Get email template from plugin configuration."""
+    plugin_configuration = PluginConfiguration.objects.filter(
+        identifier=plugin_identifier
+    ).first()
+    if not plugin_configuration:
+        return default
+    configuration = plugin_configuration.configuration
+    for config_field in configuration:
+        if config_field["name"] == template_field_name:
+            return config_field["value"] or default
+    return default
+
+
+def get_email_template_or_default(
+    plugin_identifier: str,
+    template_field_name: str,
+    default_template_file_name: str,
+    default_template_path: str,
+):
+    email_template_str = get_email_template(
+        plugin_identifier=plugin_identifier,
+        template_field_name=template_field_name,
+        default=DEFAULT_EMAIL_VALUE,
+    )
+    if email_template_str == DEFAULT_EMAIL_VALUE:
+        email_template_str = get_default_email_template(
+            default_template_file_name, default_template_path
+        )
+    return email_template_str
+
+
+def get_email_subject(
+    plugin_identifier: str, subject_field_name: str, default: str
+) -> str:
+    """Get email subject from plugin configuration."""
+    plugin_configuration = PluginConfiguration.objects.filter(
+        identifier=plugin_identifier
+    ).first()
```

Comment:
Aren't we fetching this twice? The same query is used in `get_email_template` and if I'm correct these methods are used together. We could fetch that once and pass it as an argument I think.